src/Deanery.java

```diff
@@ -0,0 +1,111 @@
+import java.util.ArrayList;
+import java.util.Optional;
+import java.util.Scanner;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDateTime;
+
+public class Deanery {
+    static DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd.MM.yyyy HH:mm:ss");
+    static LocalDateTime now = LocalDateTime.now();
+
+    private static String createRaport(Student student, String roomNumber) {
+        return "------------------------------------------" +
+                "\n" +
+                String.format("%20s", "STUDENT ") + student.getID().toString() +
+                "\n" + "\n" +
+                "First Name" + String.format("%10s", " ") + "|" + String.format("%3s", " ") + student.getName() + "\n" +
+                "Second Name" + String.format("%9s", " ") + "|" + String.format("%3s", " ") + student.getSecondName() + "\n" + "\n" +
+                "Raport generated on " + dtf.format(now) + "\n" +
+                "Room number: " + roomNumber +
+                "\n" +
+                "------------------------------------------";
+    }
+
+    public static void serve(Student student, String roomNumber, int hour) {
+        if (!(roomNumber.equals(Rooms.A123.toString())||roomNumber.equals(Rooms.B123.toString())||roomNumber.equals(Rooms.C124.toString())||roomNumber.equals(Rooms.D124.toString()))) {
+            throw new RuntimeException("The room does not match any od Deanery's rooms");
+        }
+        if (Rooms.valueOf(roomNumber) == Rooms.A123) {
+            if (student.getTypeOfStudy() == "FULL_TIME") {
+                if (hour >= Rooms.A123.getRoomHour().getStartHour() && hour <= Rooms.A123.getRoomHour().getEndHour()) {
+                    System.out.println(createRaport(student, roomNumber));
+                }
+                else {
+                    throw new RuntimeException("Wrong hours to serve student!");
```

Comment:
a Runtime exception is fine - but I wanted you to experiment a bit with implementing a custom exception 
e.g. 
_public class WrongHoursException extends Exception (or RuntimeException if you want it checked) {
    public WrongHoursException() {
        super("Wrong hours to serve a student");
   }
}_